### PR TITLE
Update `Bridge` contract to new design

### PR DESCRIPTION
**Pull request by okkothejawa**
Introduces new functions to `Bridge`:
- `markMaliciousOperator`
- `declareWithdrawFiller`

Changes `deposit`, `initialize` and `withdraw` according to the new bridge design. Also does renamings that use better nomenclature and refactoring for better code structure.

### CODE REVIEW

File: crates/evm/src/evm/system_contracts/mod.rs [hunk 1]
```diff
@@ -84,29 +84,22 @@ impl Bridge {
         // This is equal to:
         //
         // BridgeContract::initializeCall {
-        //     _depositScript: hex!("d2205daf577048c5e5a9a75d0a924ed03e226c3304f4a2f01c65ca1dab73522e6b8bad206228eba653cf1819bcfc1bc858630e5ae373eec1a9924322a5fe8445c5e76027ad201521d65f64be3f71b71ca462220f13c77b251027f6ca443a483353a96fbce222ad200fabeed269694ee83d9b3343a571202e68af65d05feda61dbed0c4bdb256a6eaad2000326d6f721c03dc5f1d8817d8f8ee890a95a2eeda0d4d9a01b1cc9b7b1b724dac00630663697472656114").into(),
-        //     _scriptSuffix: hex!("0800000000000f424068").into(),
-        //     _requiredSigsCount: U256::from(5),
+        //     _scriptPrefix: hex!("4a20a9dc2892380bf8245eb7a677ab166c98514ea6c5ce7946adbf4c05afb2bfbffdac00630663697472656114").into(),
+        //     _scriptSuffix: hex!("08000000003b9aca0068").into(),
+        //     _depositAmount: U256::from(10 ether),
```
Comment (kpp) [inline crates/evm/src/evm/system_contracts/mod.rs:89]: What is ether in the context?

Comment (eyusufatik) [reply crates/evm/src/evm/system_contracts/mod.rs:89]: ether -- it's a unit in evm 

mean x 10 ** 18

Comment (okkothejawa) [reply crates/evm/src/evm/system_contracts/mod.rs:89]: I can write directly 10 * 10 ** 18 if you want, I didn't do it as it's a comment

**Review by eyusufatik — APPROVED** [review]
LGTM
let's wait for @ekrembal 's aproval

File: crates/evm/src/evm/system_contracts/src/Bridge.sol [hunk 2]
```diff
@@ -165,7 +185,76 @@ contract Bridge is Ownable2StepUpgradeable {
...
+    /// @param slashOrTakeTp Transaction parameters of the slashOrTake transaction on Bitcoin
+    function markMaliciousOperator(TransactionParams calldata slashOrTakeTp) external {
+        validateAndCheckInclusion(slashOrTakeTp);
+        
+        (, uint256 nOuts) = BTCUtils.parseVarInt(slashOrTakeTp.vout);
+        bytes memory output = BTCUtils.extractOutputAtIndex(slashOrTakeTp.vout, nOuts - 1);
+        bytes memory opReturnData = BTCUtils.extractOpReturnData(output);
+        bytes32 moveTxId = bytesToBytes32(opReturnData.slice(0, 32));
+        uint256 operatorId = uint256(bytesToBytes32(opReturnData.slice(32, opReturnData.length - 32)));
+        uint256 depositId = txIdToDepositId[moveTxId];
```
Comment (ekrembal) [inline crates/evm/src/evm/system_contracts/src/Bridge.sol:219]: We should raise error in here if txIdToDepositId[moveTxId] is empty.
And also there is a mismatch between depositId and withdrawalUTXOs, depositId starts from 1 but withdrawalUTXOs starts from 0

**Review by ekrembal — CHANGES_REQUESTED** [review]

**Review by ekrembal — APPROVED** [review]
Tested end2end for deposit

**Pull request merged**